Tella/Scenes/Authentication/Views/Unlock/UnlockView.swift

```diff
@@ -0,0 +1,162 @@
+                        .foregroundColor(.white)
+                        .lineSpacing(7)
+                        .multilineTextAlignment(.center)
+                        .padding(EdgeInsets(top: 0, leading: 67, bottom: 0, trailing: 67))
+                    Spacer()
+                }
+                
+                
+
+                if(type == .tellaPassword) {
+                    PasswordTextFieldView(fieldContent: $viewModel.loginPassword,
+                                          isValid: .constant(true),
+                                          shouldShowError: $viewModel.shouldShowUnlockError) {
+                        loginActions()
+                    }
+                } else {
+                    PasswordTextFieldView(fieldContent: $viewModel.loginPassword,
+                                          isValid: .constant(true),
+                                          shouldShowError: $viewModel.shouldShowUnlockError,
+                                          disabled: true)
+                    
+                    Spacer(minLength: 20)
+                    
+                    PinView(fieldContent: $viewModel.loginPassword,
+                            keyboardNumbers: UnlockKeyboardNumbers) {
+                        loginActions()
+                    }
+                }
+                
+                
+                
+                Spacer()
+            }
+            
+            if  isLoading {
+                CircularActivityIndicatory()
+            }
+            
+        }
+        
+        .overlay(lockChoiceView)
+        .onAppear {
+            viewModel.initUnlockData()
+        }
+    }
+    
+    var titleString : String {
+        if(type == .tellaPin) {
+            if viewModel.shouldShowUnlockError {
+                return  LocalizableLock.unlockUpdatePinErrorIncorrectPIN.localized
+            } else {
+                return viewModel.unlockType == .new ? LocalizableLock.unlockPinSubhead.localized : LocalizableLock.unlockUpdatePinSubhead.localized
+            }
+        }
+        
+        if viewModel.shouldShowUnlockError {
+            return  LocalizableLock.unlockUpdatePasswordErrorIncorrectPassword.localized
+        } else {
+            return viewModel.unlockType == .new ? LocalizableLock.unlockPasswordSubhead.localized : LocalizableLock.unlockUpdatePasswordSubhead.localized
+        }
+    }
+    
+    var lockChoiceView : some View {
+        presentingLockChoice ? LockChoiceView( isPresented: $presentingLockChoice) : nil
+    }
+    
+    private func loginActions() {
+        viewModel.login()
+        if !viewModel.shouldShowUnlockError {
+            successLogin()
+        } else {
+            checkUnlockAttempts()
+        }
+    }
+    
+    private func successLogin() {
+        viewModel.unlockAttempts = 0
+        UserDefaults.standard.set(viewModel.unlockAttempts, forKey: "com.tella.lock.attempts")
```

Comment:
@ValbuenaG  Could you add the unlock Attempts to settings model

Reply:
what would be the purpose of adding UnlockAttempts to the settings model? We are using this only on the LockModel for specific lock view behavior. Is it worth adding it to Settings? Just double checking in case I missed something

Reply:
In my opinion it cannot be added separately, to centralise the informations in one place